yt_dlp/extractor/snapchat.py

```diff
@@ -0,0 +1,77 @@
+from .common import InfoExtractor
+from ..utils import float_or_none, int_or_none, url_or_none
+from ..utils.traversal import traverse_obj
+
+
+class SnapchatSpotlightIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?snapchat\.com/spotlight/(?P<id>[\w]+)'
+
+    _TESTS = [{
+        'url': 'https://www.snapchat.com/spotlight/W7_EDlXWTBiXAEEniNoMPwAAYYWtidGhudGZpAX1TKn0JAX1TKnXJAAAAAA',
+        'md5': '46c580f63592d0cbb76e974d2f9f0fcc',
+        'info_dict': {
+            'id': 'W7_EDlXWTBiXAEEniNoMPwAAYYWtidGhudGZpAX1TKn0JAX1TKnXJAAAAAA',
+            'ext': 'mp4',
+            'title': 'Views 💕',
+            'description': '',
+            'thumbnail': r're:https://cf-st\.sc-cdn\.net/d/kKJHIR1QAznRKK9jgYYDq\.256\.IRZXSOY',
+            'duration': 4.665,
+            'timestamp': 1637777831.369,
+            'upload_date': '20211124',
+            'repost_count': int,
+            'uploader': 'shreypatel57',
+            'uploader_url': 'https://www.snapchat.com/add/shreypatel57',
+        },
+    }, {
+        'url': 'https://www.snapchat.com/spotlight/W7_EDlXWTBiXAEEniNoMPwAAYcnVjYWdwcGV1AZEaIYn5AZEaIYnrAAAAAQ',
+        'md5': '4cd9626458c1a0e3e6dbe72c544a9ec2',
+        'info_dict': {
+            'id': 'W7_EDlXWTBiXAEEniNoMPwAAYcnVjYWdwcGV1AZEaIYn5AZEaIYnrAAAAAQ',
+            'ext': 'mp4',
+            'title': 'Spotlight Snap',
+            'description': 'How he flirt her teacher🤭🤭🤩😍 #kdrama#cdrama #dramaclips #dramaspotlight',
+            'thumbnail': r're:https://cf-st\.sc-cdn\.net/i/ztfr6xFs0FOcFhwVczWfj\.256\.IRZXSOY',
+            'duration': 10.91,
+            'timestamp': 1722720291.307,
+            'upload_date': '20240803',
+            'view_count': int,
+            'repost_count': int,
+            'uploader': 'ganda0535',
+            'uploader_url': 'https://www.snapchat.com/add/ganda0535',
+            'tags': ['#dramaspotlight', '#dramaclips', '#cdrama', '#kdrama'],
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        video_data = traverse_obj(
+            self._search_nextjs_data(webpage, video_id), (
+                'props', 'pageProps', (
+                    ('spotlightFeed', 'spotlightStories', lambda _, v: v['story']['storyId']['value'] == video_id, 'metadata', any),
+                    None), any))
```

Comment:
I would use a temp var just for readability, let the keyerror rip for `pageProps`, and use 2 separate traversal paths instead doing the extra branching with `any`
```suggestion
        page_props = self._search_nextjs_data(webpage, video_id)['props']['pageProps']
        video_data = traverse_obj(page_props, (
            'spotlightFeed', 'spotlightStories',
            lambda _, v: v['story']['storyId']['value'] == video_id, 'metadata', any), None)
```